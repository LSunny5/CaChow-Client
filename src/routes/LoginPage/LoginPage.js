import React from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import '../../components/LoginForm/LoginForm.css';
import PropTypes from 'prop-types';

class LoginPage extends React.Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => { },
        },
    }

    handleLoginSuccess = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || '/search';
        history.push(destination);
    }

    render() {
        return (
            <section className='formPage lForm'>
                <button
                    className="backButton"
                    onClick={() => this.props.history.goBack()}>
                    <img src="/images/backarrow.png" alt="Go back" className="backArrow" />
                    Back
                </button>
                <LoginForm onLoginSuccess={this.handleLoginSuccess} />
                <div className="demoInstructions">
                    <p className="makeBold">For Demo Version: </p>
                    Username: Guest
                    <br />
                    Password: Demo2020!
                </div>
            </section>
        )
    }
}

LoginPage.propTypes = {
    location: PropTypes.object,
    history: PropTypes.object,
};

export default LoginPage;