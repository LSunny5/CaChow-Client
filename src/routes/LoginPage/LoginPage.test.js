import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './LoginPage';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`LoginPage route`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        const props = {
            location: {},
            history: {
                push: () => { }
            }
        };
        ReactDOM.render(
            <BrowserRouter>
                <LoginPage {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the LoginPage form', () => {
        const wrapper = shallow(<LoginPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});