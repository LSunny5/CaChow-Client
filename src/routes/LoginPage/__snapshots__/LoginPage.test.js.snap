// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage route renders the LoginPage form 1`] = `
<section
  className="formPage lForm"
>
  <button
    className="backButton"
    onClick={[Function]}
  >
    <img
      alt="Go back"
      className="backArrow"
      src="/images/backarrow.png"
    />
    Back
  </button>
  <LoginForm
    onLoginSuccess={[Function]}
  />
  <div
    className="demoInstructions"
  >
    <p
      className="makeBold"
    >
      For Demo Version: 
    </p>
    Username: Guest
    <br />
    Password: Demo2020!
  </div>
</section>
`;
